server/container_create.go

```diff
@@ -229,13 +229,17 @@ func setupContainerUser(ctx context.Context, specgen *generate.Generator, rootfs
 	}
```

Comment:
I would expect https://github.com/cri-o/cri-o/pull/8251/files#diff-1ec6bb9cf833fd4ded46ce1feb6440f1d7532294e0445623f2f2e0bb2726ea63R226 to take the RunAsGroup field into account and then we could just rely on gid from this point on